core/src/main/scala/org/apache/spark/sql/catalyst/catalog/TiCompositeSessionCatalog.scala

```diff
@@ -0,0 +1,344 @@
+    newName.database.map(formatDatabaseName).foreach { newDb =>
+      if (db != newDb) {
+        throw new AnalysisException(
+          s"RENAME TABLE source and destination databases do not match: '$db' != '$newDb'"
+        )
+      }
+    }
+    catalogOf(oldName.database)
+      .getOrElse(throw new NoSuchDatabaseException(oldName.database.getOrElse(getCurrentDatabase)))
+      .renameTable(oldName, newName)
+  }
+
+  override def dropTable(name: TableIdentifier, ignoreIfNotExists: Boolean, purge: Boolean): Unit =
+    catalogOf(name.database)
+      .getOrElse(throw new NoSuchDatabaseException(name.database.getOrElse(getCurrentDatabase)))
+      .dropTable(name, ignoreIfNotExists, purge)
+
+  override def lookupRelation(name: TableIdentifier): LogicalPlan =
+    catalogOf(name.database)
+      .getOrElse(throw new NoSuchDatabaseException(name.database.getOrElse(getCurrentDatabase)))
+      .lookupRelation(name)
+
+  override def listTables(db: String): Seq[TableIdentifier] = listTables(db, "*")
+
+  override def listTables(db: String, pattern: String): Seq[TableIdentifier] = {
+    val currentSessionCatalog = catalogOf(Some(db)).getOrElse(throw new NoSuchDatabaseException(db))
+    val tables = currentSessionCatalog.listTables(db, pattern)
+    // list tempViews if catalog matches CH Catalog
+    val extraLocalTempViews = currentSessionCatalog match {
+      case _: TiConcreteSessionCatalog =>
+        legacyCatalog
+          .listTables(legacyCatalog.getCurrentDatabase, pattern)
+          .filter(_.database.isEmpty)
+      case _ => Seq()
+    }
+    tables ++ extraLocalTempViews
+  }
+
+  // Following are all routed to primary catalog.
+  override def createDatabase(dbDefinition: CatalogDatabase, ignoreIfExists: Boolean): Unit =
+    primaryCatalog.createDatabase(dbDefinition, ignoreIfExists)
+
+  override def getCachedPlan(t: QualifiedTableName, c: Callable[LogicalPlan]): LogicalPlan =
+    primaryCatalog.getCachedPlan(t, c)
+
+  override def getCachedTable(key: QualifiedTableName): LogicalPlan =
+    primaryCatalog.getCachedTable(key)
+
+  override def cacheTable(t: QualifiedTableName, l: LogicalPlan): Unit =
+    primaryCatalog.cacheTable(t, l)
+
+  override def invalidateCachedTable(key: QualifiedTableName): Unit =
+    primaryCatalog.invalidateCachedTable(key)
+
+  override def invalidateAllCachedTables(): Unit = primaryCatalog.invalidateAllCachedTables()
+
+  override def getDefaultDBPath(db: String): URI = primaryCatalog.getDefaultDBPath(db)
+
+  override def createTable(tableDefinition: CatalogTable, ignoreIfExists: Boolean): Unit =
+    primaryCatalog.createTable(tableDefinition, ignoreIfExists)
+
+  override def loadTable(name: TableIdentifier,
+                         loadPath: String,
+                         isOverwrite: Boolean,
+                         isSrcLocal: Boolean): Unit =
+    primaryCatalog.loadTable(name, loadPath, isOverwrite, isSrcLocal)
```

Comment:
What exactly is loadTable for?

Reply:
It only implements functions in the abstract class.